frontend/src/container/JoinWorkspace.jsx

```diff
@@ -0,0 +1,181 @@
+  }
+
+  async loadWorkspaceSubscriptions () {
+    const { props } = this
+
+    const fetchSubscriptionList = await props.dispatch(getWorkspaceSubscriptions(props.user.userId))
+
+    if (fetchSubscriptionList.status !== 200) {
+      props.dispatch(newFlashMessage(props.t('An error has happened'), 'warning'))
+      return
+    }
+    props.dispatch(setWorkspaceSubscriptionList(fetchSubscriptionList.json))
+  }
+
+  handleAllAppChangeLanguage () {
+    const { props } = this
+    this.buildBreadcrumbs()
+    props.dispatch(setHeadTitle(props.t('Join a space')))
+  }
+
+  buildBreadcrumbs () {
+    const { props } = this
+
+    props.dispatch(setBreadcrumbs([{
+      link: <Link to={PAGE.HOME}><i className='fa fa-home' />{props.t('Home')}</Link>,
+      type: BREADCRUMBS_TYPE.CORE
+    }, {
+      link: <Link to={PAGE.JOIN_WORKSPACE}>{props.t('Join a space')}</Link>,
+      type: BREADCRUMBS_TYPE.CORE
+    }]))
+  }
+
+  async joinWorkspace (workspaceId) {
+    const { props } = this
+    await props.dispatch(joinWorkspace(workspaceId, props.user.userId))
+    props.history.push(PAGE.WORKSPACE.DASHBOARD(workspaceId))
+  }
+
+  createRequestComponent (workspace) {
+    const { props } = this
+
+    const subscription = props.workspaceSubscriptionList.find(s => s.workspace.workspace_id === workspace.id)
+
+    if (subscription !== undefined) {
+      var text = 'Unknown request state'
+      var icon = 'question'
+      switch (subscription.state) {
+        case SUBSCRIPTION_TYPE.pending.slug:
+          text = props.t('Request sent')
+          icon = SUBSCRIPTION_TYPE.pending.faIcon
+          break
+        case SUBSCRIPTION_TYPE.rejected.slug:
+          text = props.t('Request rejected')
+          icon = SUBSCRIPTION_TYPE.rejected.faIcon
+          break
+      }
+      return <div><i class={`fa fa-${icon}`} /> {text}</div>
+    }
+
+    switch (workspace.accessType) {
+      case SPACE_TYPE.onRequest.slug:
+        return (
+          <IconButton
+            icon='share'
+            text={props.t('Request access')}
+            onClick={() => props.dispatch(subscribeToWorkspace(workspace.id, props.user.userId))}
+          />)
+      case SPACE_TYPE.open.slug:
+        return (
+          <IconButton
+            icon='sign-in'
+            text={props.t('Join the space')}
+            onClick={() => this.joinWorkspace(workspace.id)}
+          />)
+      default:
+        return 'Unknown space access type'
+    }
+  }
+
+  createIconForAccessType (accessType) {
+    const spaceType = SPACE_TYPE_LIST.find(t => t.slug === accessType)
+    return spaceType
+      ? <i class={`fa fa-fw fa-2x fa-${spaceType.faIcon}`} title={spaceType.label} />
+      : <i class='fa fa-fw fa-2x fa-search' title='Unknown space type' />
+  }
+
+  handleWorkspaceFilter (filter) {
+    const re = new RegExp(escapeRegExp(filter), 'i')
+    const filterFunc = workspace => workspace.label.search(re) >= 0 || workspace.description.search(re) >= 0
```

Comment:
I'm not sure what this exactly does but, if it is only to search if a workspace label or description contains the string `filter` you can use the function `.includes()` on a string.
It would avoid having to use regex

Reply:
`workspace.label.toLowerCase().includes(filter.toLowerCase())` for case insensitivity.